### DISCUSSION TITLE
Number of trainable params in network

### DISCUSSION THREAD

**Discussion by sravanbabuiitm** [dmlc/gluon-nlp#334] [Q&A] [answered]
Hi,

print(net) is giving the overview of the network, but doesnt give me the number of params that are being updated in each layer. for example, keras net.summary() gives the overview of shapes and number of params in each layer that could be trained/updated.

is there a way to figure this out from gluon API ?

**Comment by szha** [comment, marked as answer]
Gluon now supports a summary view too. https://mxnet.incubator.apache.org/versions/master/api/python/gluon/gluon.html?highlight=summary#mxnet.gluon.Block.summary
The input to the method is the typical input such as a batch.

**Comment by sravanbabuiitm** [comment]
this does require an input i wud assume ?
Since we wud be doing .hybridize(), is there a simpler way for us to view the summary from the computed static graph?

**Comment by szha** [comment]
yes, it requires an input. in order to get intermediate sizes, an input would always be needed even if it's from a static graph since gluon allows partial shape and deferred initialization.

**Comment by szha** [comment]
Closing now. Let me know if you have more questions along the line.
